File: 백준/Silver/1182. 부분수열의 합/부분수열의 합.py
import sys
data_temp = sys.stdin if sys.platform == 'linux' else open('입력.txt', 'r')
input_data = data_temp.read().splitlines()

def solution (data) :
  arr = [list(map(int, v.split())) for v in data]
  [n,s],numbers = arr[0],arr[1]
  result = 0
  
  def DFS (idx, sum) :
    nonlocal result
    
    if idx >= n : return
    sum += numbers[idx]
    if sum == s : result += 1
    
    DFS(idx+1, sum) # 현재 수 포함
    DFS(idx+1, sum - numbers[idx]) # 현재 수 배제
    
  DFS(0,0)  
  print(result)
    
solution(input_data)

File: 백준/Silver/14225. 부분수열의 합/부분수열의 합.js
const { verify } = require("crypto");
const fs = require("fs");
const input = fs
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "입력.txt")
  .toString()
  .trim()
  .split("\n");

function solution(data) {
  const [[N], arr] = data.map((el) => el.split(" ").map(Number));

  const table = {};

  function DFS(sum, cnt) {
    if (cnt === N) return;

    sum += arr[cnt];
    table[sum] = true;
    DFS(sum, cnt + 1);
    DFS(sum - arr[cnt], cnt + 1);
  }

  DFS(0, 0);

  let idx = 1;
  while (true) {
    if (!table[idx]) {
      console.log(idx);
      return;
    }
    idx++;
  }
}

solution(input);
